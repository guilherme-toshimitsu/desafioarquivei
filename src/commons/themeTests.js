import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import * as Redux from "react-redux";
import thunk from "redux-thunk";
import theme from "./theme";
import store from "@store";

const renderWithTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};

const renderWithThemeAndRedux = (children) => {
  return render(
    <ThemeProvider theme={theme}>
      <Redux.Provider store={store}>{children}</Redux.Provider>
    </ThemeProvider>
  );
};

export { renderWithTheme, renderWithThemeAndRedux };
