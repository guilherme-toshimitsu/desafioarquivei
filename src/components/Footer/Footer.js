import { memo } from "react";
import styled from "styled-components";
import { breakpoints } from "@commons/theme";

const StyledButton = styled.footer`
  position: fixed;
  left: 0;
  bottom: 0;
  width: 100%;
  color: white;
  height: 40px;
  background-color: ${({ theme }) => theme.colors.main};
  ${breakpoints.sm} {
    height: 20px;
  }
`;

export default memo(StyledButton);
