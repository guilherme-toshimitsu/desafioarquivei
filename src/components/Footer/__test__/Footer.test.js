import React from "react";
import Footer from "..";
import { renderWithTheme } from "@commons/themeTests";

describe("<Footer />", () => {
  it("creates a Footer component", () => {
    const wrapper = renderWithTheme(<Footer />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
