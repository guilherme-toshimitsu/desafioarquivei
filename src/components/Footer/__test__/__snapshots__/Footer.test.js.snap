// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> creates a Footer component 1`] = `
<DocumentFragment>
  <footer
    class="Footer__StyledButton-zgi4km-0 bSKKpt"
  />
</DocumentFragment>
`;
